src/rust_api.rs

```diff
@@ -143,6 +152,45 @@ pub struct DnaSequence<T: NucleotideLike> {
     dna: Vec<T>,
 }
 
+#[cfg(feature = "serde")]
+impl<'de, T: NucleotideLike> serde::Deserialize<'de> for DnaSequence<T> {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        T: NucleotideLike,
+        D: serde::Deserializer<'de>,
+    {
+        struct Helper<T>(PhantomData<T>);
```

Comment:
Given my inexperience with `serde` I worry I may be putting my foot in my mouth, but could you improve the DRYness by pulling out the inner `Helper<T>` from the `Deserialize` impl of `DnaSequence<T>` and storing it in `src/serde_utils.rs`, then making the `impl_stringlike!` macro use it? Even though it wouldn't be the same as having `impl_stringlike!` support generic types, it'd at least only have one copy of the `Helper` and shrink the `Deserialize` impl for `DnaSequence<T>` down to something like:

```rust
#[cfg(feature = "serde")]
impl<'de, T: NucleotideLike> serde::Deserialize<'de> for DnaSequence<T> {
    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
    where
        T: NucleotideLike,
        D: serde::Deserializer<'de>,
    {
        deserializer.deserialize_str(StringlikeVisitor(PhantomData))
    }
}
```